inc/status.h

```diff
@@ -61,6 +61,8 @@ typedef struct Status {
   const char* message;
 } Status;
 
+typedef void (*status_callback)(Status *const status);
```

Comment:
Pretty sure you were right the first time.
In this case, this is a constant pointer to data.
We probably want a pointer to constant data.